File: package/main/src/Random/randomChoice.ts
/**
 * Returns a uniformly random element from the input array.
 * @template T - Element type
 * @param {readonly T[]} items - Source array (must be non-empty)
 * @returns {T} A randomly selected element
 * @example
 * randomChoice(["a", "b", "c"]);
 */
export const randomChoice = <T>(items: readonly T[]): T => {
  const index = Math.floor(Math.random() * items.length);
  return items[index];
};

File: package/umt_python/src/random/random_choice.py
import math
import random as random_module
from collections.abc import Sequence
from typing import TypeVar

T = TypeVar("T")


def random_choice(items: Sequence[T]) -> T:
    """
    Returns a uniformly random element from the input sequence.

    Args:
        items: Source sequence (must be non-empty).

    Returns:
        A randomly selected element.

    Example:
        >>> random_choice([42])
        42
        >>> random_choice(["a", "b", "c"]) in ("a", "b", "c")
        True
    """
    index = math.floor(random_module.random() * len(items))
    return items[index]
